src/Compilers/CSharp/Portable/Compilation/InitializerSemanticModel.cs

```diff
@@ -181,20 +181,22 @@ private BoundNode BindEqualsValue(Binder binder, EqualsValueClauseSyntax equalsV
             {
                 case SymbolKind.Field:
                     {
-                        var enumField = this.MemberSymbol as SourceEnumConstantSymbol;
+                        var fieldSymbol = (FieldSymbol)this.MemberSymbol;
+                        var enumField = fieldSymbol as SourceEnumConstantSymbol;
                         if ((object)enumField != null)
                         {
                             return binder.BindEnumConstantInitializer(enumField, equalsValue.Value, diagnostics);
                         }
 
-                        var fieldType = ((FieldSymbol)this.MemberSymbol).GetFieldType(binder.FieldsBeingBound);
-                        return binder.BindVariableOrAutoPropInitializer(equalsValue, fieldType, diagnostics);
+                        var fieldType = (fieldSymbol).GetFieldType(binder.FieldsBeingBound);
+                        return new BoundFieldInitializer(equalsValue, fieldSymbol, binder.BindVariableOrAutoPropInitializer(equalsValue, fieldType, diagnostics));
                     }
 
                 case SymbolKind.Property:
                     {
-                        var propertyType = ((PropertySymbol)this.MemberSymbol).Type;
-                        return binder.BindVariableOrAutoPropInitializer(equalsValue, propertyType, diagnostics);
+                        var propertySymbol = (PropertySymbol)this.MemberSymbol;
+                        var propertyType = propertySymbol.Type;
+                        return new BoundPropertyInitializer(equalsValue, propertySymbol, binder.BindVariableOrAutoPropInitializer(equalsValue, propertyType, diagnostics));
```

Comment:
I added BoundPropertyInitializer for symmetry. It isn't possible to exercise this path with the patterns of the analyzer I was writing, but this code path is apparently live--when I inserted an exception throw here a couple of tests failed. I'm a bit uncomfortable knowing that this code can be reached by some means other than analyzing code blocks for member initializers--and not knowing precisely what those means are--but all tests pass with this change.